libcxx/test/std/thread/thread.mutex/thread.lock/thread.lock.shared/thread.lock.shared.cons/move_assign.pass.cpp

```diff
@@ -35,8 +33,13 @@ int main(int, char**)
     assert(lk1.owns_lock() == true);
     assert(lk0.mutex() == nullptr);
     assert(lk0.owns_lock() == false);
-    }
-    {
+
+    // Test self move assignment.
+    lk1 = std::move(lk1);
+    assert(lk1.mutex() == std::addressof(m0));
+    assert(lk1.owns_lock());
+  }
+  {
```

Comment:
I noticed this test does not validate the `noexcept` status of the function, can you add that.
(The next test should add it since there it's a new requirement.)
The test for `swap` has an example how to test for `noexcept`.